### DISCUSSION TITLE
Composer 1.10.22: --no-dev wants to include require-dev

### DISCUSSION THREAD

**Discussion by Crizz0** [composer/composer#9859] [Q&A / Support]
Hi,

did run:
composer.phar update --no-dev

with a composer.json which contains:
```
	"require-dev": {
		"phpbb/epv": "dev-master"
	},
[...]
	"config": {
		"platform":{
			"php":"5.6"
		}
	}
```

Which led to:
![no-dev-conflict](https://user-images.githubusercontent.com/1311778/116687794-018b9900-a9b6-11eb-9925-68f3afea64cf.PNG)

I thought, this require-dev would be ignored. But it doesn't. 

Did i understand the command wrong? 

Regards

**Comment by stof** [comment]
`--no-dev` means that the dev dependencies won't be *installed*. It does not mean that they are not resolved. Otherwise, the locked version for non-dev dependencies could change as well (as it would suddenly allow you to select a version for them incompatible with dev dependencies) and would make the generated lock file unusable for anyone running `composer install` without ``--no-dev`.
